substrate-archive/src/database/listener.rs

```diff
@@ -214,21 +224,31 @@ impl Listener {
 	}
 }
 
+impl Drop for Listener {
+	fn drop(&mut self) {
+		if let Err(e) = task::block_on(self.kill()) {
+			log::error!("{}", e)
```

Comment:
I think you could add some more context here, such as `"could not terminate task <TASK> {}", error`